### PR TITLE
[IMP] core: deprecate odoo.registry()

### PR DESCRIPTION
**Pull request by kmagusiak**
You can use directly `odoo.modules.registry.Registry` when needed and cases where you need to do it are rare. At the same time, this allows to remove a function from the `odoo` package.

Related PRs:
Enterprise: odoo/enterprise#69269
Upgrade: odoo/upgrade-util#132

Task [link](https://www.odoo.com/odoo/project/1364/tasks/4069446)
task-4069446

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by lordkrandel — CHANGES_REQUESTED** [review]
Some minor

File: addons/website/models/website.py [hunk 1]
```diff
@@ -89,6 +90,11 @@
 ])
 
 
+def registry():
+    database_name = threading.current_thread().dbname
+    return Registry(database_name)
```
[the remark this answers is not in the document]
Comment (kmagusiak) [reply addons/website/models/website.py:95]: After checking the code, they can be removed as the registry is already reset in `_button_immediate_function`. A similar assert is in there.

File: odoo/__init__.py [hunk 2]
```diff
@@ -33,6 +35,7 @@ def registry(database_name=None):
     on the current thread. If the registry does not exist yet, it is created on
     the fly.
     """
+    warnings.warn("Use directly odoo.modules.registry.Registry", DeprecationWarning)
```
Comment (lordkrandel) [inline odoo/__init__.py:38]:
```suggestion
    from warnings import warn  # noqa: PLC0415
    warn("Use directly odoo.modules.registry.Registry", DeprecationWarning)
```

File: odoo/__init__.py [hunk 3]
```diff
@@ -23,6 +23,8 @@
 # ----------------------------------------------------------
 # Shortcuts
 # ----------------------------------------------------------
+import warnings
+
```
Comment (lordkrandel) [inline odoo/__init__.py:27]:
```suggestion
```

**Review by lordkrandel — CHANGES_REQUESTED** [review]
I wouldn't add `warnings` to the `odoo` namespace, it's a very special one.

**Comment by lordkrandel** [comment]
As far as the code here, I have no more comments.

For the PR scope, I'm probably not the best person to judge, but since the task is kinda explicit on what to do, I would also assume it's ok. 

Someone would have to verify that all the codepoints have been fixed, and some manual testing that nothing blows up, even though automated testing should already do.

You can add the related PRs also to the commit message, for fast reference.

**Review by ryv-odoo — APPROVED** [review]
@robodoo r+
LGTM

**Pull request closed without merging**